LogicalTypes.md

```diff
@@ -563,6 +563,41 @@ defined by the [BSON specification][bson-spec].
 
 The sort order used for `BSON` is unsigned byte-wise comparison.
 
+### VARIANT
+
+`VARIANT` is used for a Variant value. It must annotate a group. The group must
+contain a `binary` field named `metadata`, and a `binary` field named `value`.
```

Comment:
We have been using `BYTE_ARRAY` instead of `binary` in this doc.

Reply:
Ah, you're right. The type is `BYTE_ARRAY` in thrift but `binary` in actual type definitions.

I think that `binary` is more clear, but we should mention that they are synonyms at a minimum. How about this?

> The group must contain a field named `metadata` and a field named `value`. Both fields must have type `binary`, which is also called `BYTE_ARRAY` in the Parquet thrift definition.

Reply:
Sounds good! Updated.